src/main/webapp/js/searchableField.js

```diff
@@ -10,35 +29,59 @@
 Behaviour.specify(".searchable", 'searchableField', 200, function(el) {
     var results = {};
 
-    new ComboBox(el, function(query) {
+    /**
+     * This combobox comes from https://github.com/jenkinsci/jenkins/blob/master/war/src/main/webapp/scripts/combobox.js
+     */
+    var combobox = new ComboBox(el, function(query) {
         return Object.keys(results);
     }, {});
 
-    el.addEventListener('input', function(e) {
+    el.addEventListener('input', debounce(function (e) {
+        // Clear fields dependent on this
+        document.querySelectorAll('[filldependson~="' + e.target.name.replace("_.", "") + '"]').forEach(function (el) {
+            if (el.name !== e.target.name) {
+                el.value = "";
+                document.getElementById(el.getAttribute('valueField')).value = "";
+            }
+        });
+        document.getElementById(el.getAttribute('valueField')).value = "";
+
+        if (el.value.length < 2) { // Only perform a search if there are enough characters
+            results = {};
+            return;
+        }
+
+        // Get the values of the field this depends on
         var parameters = (el.getAttribute("fillDependsOn") || "").split(" ")
-            .reduce(function(params, fieldName) {
+            .reduce(function (params, fieldName) {
                 var dependentField = findNearBy(el, fieldName);
                 if (dependentField) {
                     params[fieldName] = dependentField.value;
                 }
                 return params;
             }, {});
+        // Request the search results
         new Ajax.Request(el.getAttribute("fillUrl"), {
             parameters: parameters,
-            onSuccess: function(rsp) {
+            onSuccess: function (rsp) {
                 var valueIdentifier = el.getAttribute("valueIdentifier");
                 results = (rsp.responseJSON.data && rsp.responseJSON.data.values || [])
                     .reduce(function (flattened, result) {
                         flattened[result.name] = result[valueIdentifier];
                         return flattened;
                     }, {});
+            },
+            onError: function (rsp) {
+                results = {};
             }
         });
-    });
+        combobox.showDropdown();
+    }, 200));
```

Comment:
Based on some testing (totally scientific, me applying a debounced callback to a text field and seeing how often it is called when I am typing), I think 200 might be a tad too low, 300 – 400 may be a better range. (400 felt a bit slow, 300 might be the sweet spot for me). How did you feel the 200ms delay was? Was it still triggering very frequently, or are you a faster typer :P 

What might also be useful is setting your connection speed to either "Fast 3G" or "Slow 3G" in the devtools network tab and seeing what happens when the user's connection is slower so you don't have your requests coming back instantly, that's when you'll see more benefit from the fewer requests.

![screenshot from chrome devtools network panel](https://content.screencast.com/media/9a187fcc-7c4c-4dc6-9860-c16f4802364d_293c3d78-bf1d-4d51-becd-e53646a12bfd_static_0_0_2019-09-12_15-31-53.png)